### ISSUE TITLE
Individual templates doesn't load at form generator

### ISSUE THREAD

**Issue by andreheeke** [contao/contao#828]
**Affected version(s)**
4.8.4

**Description**  
The individual templates are no longer loaded in the form generator with the field type "Textfield". 

**How to reproduce**  
Create a new template with the name `form_textfield_<NAME>.html5` and try to select it in the form generator.

**Comment by contaoacademy** [comment]
I can confirm this problem.

Maybe the changes in #731 has to do something with this behavior.

**Comment by jonnysp** [comment]
same here please fix this soon

**Comment by leofeyer** [comment]
Your template name is wrong.

 * `FormExplanation` -> `form_explanation_custom`
 * `FormHtml` -> `form_html_custom`
 * `FormFieldsetStart` -> `form_fieldsetStart_custom`
 * `FormFieldsetStop` -> `form_fieldsetStop_custom`
 * **`FormTextField` -> `form_text_custom`**
 * `FormPassword` -> `form_password_custom`
 * `FormTextArea` -> `form_textarea_custom`
 * **`FormSelectMenu` -> `form_select_custom`**
 * **`FormRadioButton` -> `form_radio_custom`**
 * `FormCheckBox` -> `form_checkbox_custom`
 * **`FormFileUpload` -> `form_upload_custom`**
 * `FormHidden` -> `form_hidden_custom`
 * `FormCaptcha` -> `form_captcha_custom`
 * `FormSubmit` -> `form_submit_custom`

I am aware that the bold mappings are confusing, however, we cannot change them until Contao 5, because it would break backwards compatibility.

**Comment by fritzmg** [comment]
> Your template name is wrong.

The core uses `form_textfield`, not `form_text`: https://github.com/contao/contao/blob/4.8.4/core-bundle/src/Resources/contao/templates/forms/form_textfield.html5

**Comment by aschempp** [comment]
Sorry about that. I think what @leofeyer means is that a custom template must be named `form_text_xxx`, even though the default name is `form_textfield`. That's because the "custom" name is derived from the field type, and the original templates are wrong since forever but can't be changed for BC reasons.

**Comment by fritzmg** [comment]
Out of the bold highlighted list, only the FormTextField is an outlier, because the original template names of the others are in line with what you can actually select.

So for instance, if you go to _Themes_ - _Templates_ and create a new `form_select` template and then rename it to `form_select_custom`, it will work.

But if you create a new `form_textfield` template and you rename it to `form_textfield_custom`, it will not work.

Previously this wasn't a problem, because you were able to select any `form_` template.

Imho the options callback should allow both `form_text_` and `form_textfield_` for the `text` field.

Otherwise this is just confusing and having to document this somehow is weird ;)

**Comment by leofeyer** [comment]
> Out of the bold highlighted list, only the FormTextField is an outlier, because the original template names of the others are in line with what you can actually select.

That is correct, however, we want the template name to derive from the class name and the other highlighted rows do not meet this requirement. I have marked the classes that need to be adjusted in d314f0be406fd88387eb9d7985f265b48458c7ed.

**Comment by fritzmg** [comment]
> That is correct, however, we want the template name to derive from the class name and the other highlighted rows do not meet this requirement.

Of course. But an administrator simply creating a new template for a text field would not even know anything about the class name. He only sees the template's name - which is `form_textfield`. And since, with every other template (including content elements & modules), you have to add `_<custom>` to the original template name, it will not be immedaitely apparent to the user, why he would not be able to select this template.

**Comment by leofeyer** [comment]
I know. But we cannot rename the template without breaking BC…

**Comment by fritzmg** [comment]
You do not have to rename the template. Only the select field in the back end needs to offer `form_textfield_...` as well, in addition to `form_text_...`.

**Comment by leofeyer** [comment]
Guess we can do this.

**Comment by fritzmg** [comment]
For example:
```php
public function getFormFieldTemplates(Contao\DataContainer $dc)
{
	if ('text' === $dc->activeRecord->type) {
		return array_merge(
			$this->getTemplateGroup('form_text_'),
			$this->getTemplateGroup('form_textfield_')
		);
	}

	return $this->getTemplateGroup('form_' . $dc->activeRecord->type . '_');
}
```
Or something like that.

**Comment by leofeyer** [comment]
Fixed in b72ffb5812041154687c62f4bfb32705e832afc3.
